tests/Module/CheckoutTest.php

```diff
@@ -0,0 +1,578 @@
+			->willReturn( [] );
+
+		$wpAdapter                    = $this->getPacketeryMockFactory()->createWpAdapter();
+		$wcAdapter                    = $this->createMock( WcAdapter::class );
+		$productEntityFactory         = $this->createMock( ProductEntityFactory::class );
+		$productCategoryEntityFactory = $this->createMock( ProductCategoryEntityFactory::class );
+		$carrierOptionsFactory        = $this->createMock( CarrierOptionsFactory::class );
+		$carrierEntityRepository      = $this->createMock( Carrier\EntityRepository::class );
+		$currencySwitcherFacade       = $this->getPacketeryMockFactory()->createCurrencySwitcherFacade();
+		$carDeliveryConfig            = $this->createMock( CarDeliveryConfig::class );
+
+		$checkout = $this->createCheckoutMock(
+			$wpAdapter,
+			$wcAdapter,
+			$productEntityFactory,
+			$productCategoryEntityFactory,
+			$carrierOptionsFactory,
+			$currencySwitcherFacade,
+			$carrierEntityRepository,
+			$carDeliveryConfig,
+		);
+
+		$wcAdapter
+			->method( 'cart' )
+			->willReturn( $cart );
+		$wcAdapter
+			->method( 'customerGetShippingCountry' )
+			->willReturn( 'cz' );
+		$wcAdapter
+			->method( 'cartGetCartContents' )
+			->willReturn( [ $cartItem ] );
+		$wcAdapter
+			->method( 'cartGetCartContentsTotal' )
+			->willReturn( 100.0 );
+		$wcAdapter
+			->method( 'cartGetCartContentsTax' )
+			->willReturn( 21.0 );
+		$wcAdapter
+			->method( 'cartGetCartContentsWeight' )
+			->willReturn( $cartWeight );
+		$wcAdapter
+			->method( 'getWeight' )
+			->willReturn( $cartWeight );
+		$wcAdapter
+			->method( 'cartGetCartContent' )
+			->willReturn( [ $cartItem ] );
+
+		$carrierEntityRepository
+			->method( 'getByCountryIncludingNonFeed' )
+			->willReturn( $carriers );
+		$productEntityFactory
+			->method( 'fromPostId' )
+			->willReturn( $product );
+
+		$carrierOptionsFactory
+			->method( 'createByOptionId' )
+			->willReturnCallback( function ( $optionId ) use ( $carriersOptions ) {
+				$carrierOptions = $carriersOptions[ $optionId ];
+
+				return new Carrier\Options(
+					$optionId,
+					$carrierOptions
+				);
+			} );
+
+		$wcAdapter
+			->method( 'productFactoryGetProduct' )
+			->willReturn( $wcProduct );
+
+		$productCategory = $this->createMock( ProductCategory\Entity::class );
+		$productCategory
+			->method( 'getDisallowedShippingRateIds' )
+			->willReturn( array_merge( [ Carrier\OptionPrefixer::getOptionId( 'anyDisallowedByProductCategory' ) ], $productCategoryDisallowedRateIds ) );
+
+		$productCategoryEntityFactory
+			->method( 'fromTermId' )
+			->willReturn( $productCategory );
+
+		$carDeliveryConfig
+			->method( 'isEnabled' )
+			->willReturn( $isCarDeliveryEnabled );
+
+		$wpAdapter
+			->expects( self::atLeast( $expectedRateCount ) )
+			->method( 'applyFilters' );
+
+		$wpAdapter
+			->method( 'didAction' )
+			->willReturn( 1 );
+
+		$rates = $checkout->getShippingRates( $allowedCarrierNames );
+
+		self::assertCount( $expectedRateCount, $rates );
```

Comment:
assert count & values pls.

Reply:
Tested method should be refactored soon. After that, this change would not be meaningful.